### PR TITLE
<MobileDrawer /> - New Component

### PR DESCRIPTION
**Pull request by ErezPaperny**
### 🔦 Summary

Create new component <MobileDrawer />
MobileDrawers provide access to destinations and functionality menus, such as Action Menus, Filter Dropdown etc. This component should be used on MOBILE ONLY.

Documents: 
- https://docs.google.com/document/d/1Mm4B0tqlD4R4TQtMTReES61R7qrVH4BwtAVT2eJwB-M/edit#
- https://zeroheight.com/7sjjzhgo2/p/447d36-06-overlays/b/08cd42/t/98687d

### ✅ Checklist

- [V] 👨‍💻 API change is approved by the UI Infra Developers @sivanharel  
- [V] 👨‍🎨 UX change is approved by the Design System UX @michalvardi  
- 📚 Change is documented
  - [V] Story
  - [V] API description
  - [V] Cheatsheet
  - [V] Other (explain)- Example of menu's / Share social mobile drawers

- 🔬 Change is tested
  - [V] Component
  - [X] Visual test - There is no visual test for modal so can't test it visually except manually

### CODE REVIEW

**Comment by WixBuildServer** [comment]
### 🤖 Hi, I'm #Janet!

  A deploy preview of your PR was successfully released 🖼.

  You can find it under one of the following URLs:
  * https://bo.wix.com/pages/wix-ui-tpa.pr_6751/

File: packages/wix-ui-tpa/src/components/MobileDrawer/MobileDrawer.tsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
+import * as React from 'react';
+import { classes } from './MobileDrawer.st.css';
```
Comment (sivanharel) [inline packages/wix-ui-tpa/src/components/MobileDrawer/MobileDrawer.tsx:2]:
```suggestion
import { st, classes } from './MobileDrawer.st.css';
```

You should use the st function

File: packages/wix-ui-tpa/src/components/MobileDrawer/MobileDrawer.visual.tsx [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import * as React from 'react';
+import { visualize, story, snap } from 'storybook-snapper';
+import { MobileDrawer } from './';
+
+visualize('MobileDrawer', () => {
+  story('simple', () => {
+    snap('default props', <MobileDrawer>Children</MobileDrawer>);
```
Comment (sivanharel) [inline packages/wix-ui-tpa/src/components/MobileDrawer/MobileDrawer.visual.tsx:7]:
```suggestion
    snap('default props', <MobileDrawer isOpen>Children</MobileDrawer>);
```

Comment (sivanharel) [reply packages/wix-ui-tpa/src/components/MobileDrawer/MobileDrawer.visual.tsx:7]: We have a mobile wrapper for the visual tests but it has issues so we don't want to use it now.
So let's just add the same examples as you created in the example.tsx file.
I mean let's pass `actionMenuLayout` and `SocialBar` as children.

**Review by sivanharel — CHANGES_REQUESTED** [review]
Very nice!
Thanks

File: packages/wix-ui-tpa/src/components/MobileDrawer/MobileDrawer.visual.tsx (hunk 2 above)
Comment (sivanharel) [reply packages/wix-ui-tpa/src/components/MobileDrawer/MobileDrawer.visual.tsx:7]: In order to see the visual test story page, you could run: `npm run eyes-storybook-debug` and click on the link `http://localhost:6007/`

**Review by ErezPaperny — COMMENTED** [review]
@sivanharel, I agree and fixed most, few comments left open with my answer. Please review and advice if more info is needed.

**Review by sivanharel — COMMENTED** [review]
Great work!
I added some comments.

We just have one issue. We need to think about another solution for updating the visible state. It couldn't run each render.

**Review by jonathanadler — COMMENTED** [review]
I think it might work without settimeout.
I'm not 100% sure, but probably something like this:

```jsx
export class MobileDrawer extends React.Component {
  state = {
    showDrawer: this.props.isOpen,
  }

  _hideDrawer() {
    const { showDrawer } = this.state;
    const { isOpen } = this.props;

    if (!isOpen && showDrawer) {
      this.setState({ showDrawer: false});
    }
  }

  render() {
    const { showDrawer } = this.state;
    const { isOpen } = this.props;
    
    return (
      <Modal isOpen={isOpen || showDrawer}>
        <div className={classnames(
                  classes.drawer, 
                  [classes.show]: showDrawer,
                )} 
                onTransitionEnd={this._hideDrawer}
                ></div>
      </Modal>
    );
  }
}
```
and the css will look like this:
```css
.drawer {
  position: absolute;
  bottom: -100%;
  left: 0;
  transition: bottom .2s ease;
}

.drawer.show {
  bottom: 0;
}
```

File: packages/wix-ui-tpa/src/components/MobileDrawer/MobileDrawer.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+      const driver = createDriver(<MobileDrawer>Cildren</MobileDrawer>);
+
+      expect(await driver.isOpen()).toBe(false);
+    });
+
+    it('expect mobileDrawer to be isOpen if open', async () => {
+      const driver = createDriver(<MobileDrawer isOpen>Cildren</MobileDrawer>);
+
+      expect(await driver.isOpen()).toBe(true);
+    });
```
[the remark this answers is not in the document]
Comment (ErezPaperny) [reply packages/wix-ui-tpa/src/components/MobileDrawer/MobileDrawer.spec.tsx]: Hi in my opinion it is part of the testkit but I will IZROM with you and move it out of the test kit describe ;-).

File: packages/wix-ui-tpa/src/components/MobileDrawer/MobileDrawer.st.css [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  -st-named: color, font, fallback;
+}
+
+.root {
+width: 100%;
+}
+
+.contentWrapper {
+-st-states: visible;
+transform: translateY(320px);
```
[the remark this answers is not in the document]
Comment (ErezPaperny) [reply packages/wix-ui-tpa/src/components/MobileDrawer/MobileDrawer.st.css]: MAGNIV, changed, 10x

File: packages/wix-ui-tpa/src/components/MobileDrawer/MobileDrawer.tsx [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+      onRequestClose,
+      className,
+    } = this.props;
+    const { visible } = this.state;
+
+    if (isOpen && !visible) {
+      setTimeout(() => {
+        this.setState({ visible: true });
+      }, 0);
+    }
```
[the remark this answers is not in the document]
Comment (ErezPaperny) [reply packages/wix-ui-tpa/src/components/MobileDrawer/MobileDrawer.tsx]: cool, with the componentDidUpdate its actually resolve the needs of the setTimeOut and its work perfectlly. Thanks for the idea,

**Review by sivanharel — APPROVED** [review]
Looks good!
I will ask Michal to review it.

The visual tests fail since it takes the screenshot before the transition end. So we need to fix it somehow :) I will take a look on it later.

**Comment by WixBuildServer** [comment]
### 🤖 Hi, I'm #Janet!

  A deploy preview of your PR was successfully released 🖼.

  Project Name: **wix-ui-tpa**

  You can find it under one of the following URLs:
  * https://bo.wix.com/pages/wix-ui-tpa.pr_6751/

**Comment by WixBuildServer** [comment]
### 🤖 Hi, I'm #Janet!

  A deploy preview of your PR was successfully released 🖼.

  Project Name: **wix-style-react**

  You can find it under one of the following URLs:
  * https://preview.wix-style-react.com/pr_6751

**Pull request merged**